psa-crypto-sys/build.rs

```diff
@@ -29,155 +29,58 @@
 
-fn get_value_from_table<'a>(table: &'a Value, key: &str) -> Result<&'a Value> {
-    match table {
-        Value::Table(table) => table.get(key).ok_or_else(|| {
-            println!("Config table does not contain configuration key: {}", key);
-            Error::new(ErrorKind::InvalidInput, "Configuration key missing.")
-        }),
-        _ => Err(Error::new(
-            ErrorKind::InvalidInput,
-            "Value provided is not a TOML table",
-        )),
-    }
-}
-
-// Get the Mbed Crypto version to branch on from Cargo.toml file. Use that and MbedConfig to pass
-// parameters to the setup_mbed_crypto.sh script which clones and builds Mbed Crypto and create
-// a static library.
-fn setup_mbed_crypto(mbed_config: &MbedConfig, mbed_version: &str) -> Result<()> {
-    let (mbed_compiler, mbed_archiver) =
-        if std::env::var("TARGET").unwrap() == "aarch64-unknown-linux-gnu" {
-            let toolchain;
-            toolchain = mbed_config
-                .aarch64_unknown_linux_gnu
-                .as_ref()
-                .ok_or_else(|| {
-                    Error::new(
-                        ErrorKind::InvalidInput,
-                        "The aarch64_unknown_linux_gnu subtable of mbed_config should exist",
-                    )
-                })?;
-            (
-                toolchain
-                    .mbed_compiler
-                    .clone()
-                    .unwrap_or_else(|| DEFAULT_ARM64_MBED_COMPILER.to_string()),
-                toolchain
-                    .mbed_archiver
-                    .clone()
-                    .unwrap_or_else(|| DEFAULT_ARM64_MBED_ARCHIVER.to_string()),
-            )
-        } else {
-            let toolchain;
-            toolchain = mbed_config.native.as_ref().ok_or_else(|| {
-                Error::new(
-                    ErrorKind::InvalidInput,
-                    "The native subtable of mbed_config should exist",
-                )
-            })?;
-            (
-                toolchain
-                    .mbed_compiler
-                    .clone()
-                    .unwrap_or_else(|| DEFAULT_NATIVE_MBED_COMPILER.to_string()),
-                toolchain
-                    .mbed_archiver
-                    .clone()
-                    .unwrap_or_else(|| DEFAULT_NATIVE_MBED_ARCHIVER.to_string()),
-            )
-        };
-
-    let script_fail = |_| {
-        Err(Error::new(
-            ErrorKind::Other,
-            "setup_mbed_crypto.sh script failed",
-        ))
-    };
-
-    println!("cargo:rerun-if-changed={}", SETUP_MBED_SCRIPT_PATH);
-    println!("cargo:rerun-if-changed={}", "src/c/Makefile");
+fn setup_mbed_crypto() -> Result<()> {
+    let mbedtls_dir = String::from("./mbedtls");
     println!("cargo:rerun-if-changed={}", "src/c/shim.c");
     println!("cargo:rerun-if-changed={}", "src/c/shim.h");
+    // The SHARED env variable tells the build to output dynamic libraries too
+    env::set_var("SHARED", "1");
```

Comment:
Following the "decision path" comment below, this would not be needed if we always link with the libs we compile ourselves statically.